psutil_agent/loop.py

```diff
@@ -0,0 +1,80 @@
+import json
+import sched
+import sys
+import time
+
+import pika
+import psutil
+
+
+def collect_metrics(rabbit_config, psutil_config):
+    scheduler = sched.scheduler(time.time, time.sleep)
+
+    credentials = pika.PlainCredentials(rabbit_config['broker_user'],
+                                        rabbit_config['broker_pass'])
+
+    params = pika.ConnectionParameters(credentials=credentials,
+                                       host=rabbit_config['broker_hostname'],
+                                       virtual_host=rabbit_config['vhost'],
+                                       port=rabbit_config['broker_port'])
+
+    connection = pika.BlockingConnection(params)
+    channel = connection.channel()
+    channel.exchange_declare(exchange='cloudify-monitoring',
+                             exchange_type="topic",
+                             auto_delete=True,
+                             durable=False,
+                             internal=False)
+
+    for config in psutil_config:
+        def create_scheduled_fun(method, interval, f_args, result_argument,
+                                 alias):
+            fun = getattr(psutil, method)
+
+            def scheduled_fun():
+                result = fun(**f_args)
+
+                if result_argument:
+                    result = getattr(result, result_argument)
+
+                service_elements = [
+                    rabbit_config['deployment_id'],
+                    rabbit_config['node_name'],
+                    rabbit_config['node_id'],
+                    alias or method
+                ]
+
+                metric_data = {
+                    'node_id': rabbit_config['node_id'],
+                    'node_name': rabbit_config['node_name'],
+                    'deployment_id': rabbit_config['deployment_id'],
+                    'name': alias or method,
+                    'path': '',
+                    'metric': result,
+                    'unit': '',
+                    'type': 'GAUGE',
+                    'host': rabbit_config['node_id'],
+                    'service': '.'.join(service_elements),
+                    'time': int(time.time()),
+                }
+
+                channel.basic_publish(
```

Comment:
which logger should I use in except? there is no context available here, so I can't get a default one.